.github/workflows/create_release.yml

```diff
@@ -30,7 +31,22 @@ jobs:
 
       - name: Build (opt)
         run: |
+          # The ref resolves to "main" for latest and e.g. "v0.1.0" for tagged.
+          REF=${{ github.ref }}
+          REF=${REF#refs/*}
+          # For branches, this is e.g. 'heads/<branch_name>', so remove 'heads/'.
+          REF=${REF#heads/*}
+          # For tags, this is e.g. 'tags/<branch_name>', so remove 'tags/'.
+          REF=${REF#tags/*}
```

Comment:
I think you can just do
REF=${GITHUB_REF#refs/*/}
which will map both /refs/heads/foo and /refs/tags/foo to foo.